ako-gateway-api/nodes/route_validator.go

```diff
@@ -52,18 +52,22 @@ func IsHTTPRouteValid(key string, obj *gatewayv1.HTTPRoute) bool {
 	var invalidParentRefCount int
 	parentRefIndexInHttpRouteStatus := 0
 	indexInCache := -1
-	for parentRefIndexFromSpec := range httpRoute.Spec.ParentRefs {
-		err := validateParentReference(key, httpRoute, httpRouteStatus, parentRefIndexFromSpec, &parentRefIndexInHttpRouteStatus, &indexInCache)
-		if err != nil {
-			invalidParentRefCount++
-			parentRefName := httpRoute.Spec.ParentRefs[parentRefIndexFromSpec].Name
-			utils.AviLog.Warnf("key: %s, msg: Parent Reference %s of HTTPRoute object %s is not valid, err: %v", key, parentRefName, httpRoute.Name, err)
+	isValidHttprouteRules := validateHTTPRouteRules(key, httpRoute, httpRouteStatus)
+	if isValidHttprouteRules {
+		for parentRefIndexFromSpec := range httpRoute.Spec.ParentRefs {
+			err := validateParentReference(key, httpRoute, httpRouteStatus, parentRefIndexFromSpec, &parentRefIndexInHttpRouteStatus, &indexInCache)
+			if err != nil {
+				invalidParentRefCount++
+				parentRefName := httpRoute.Spec.ParentRefs[parentRefIndexFromSpec].Name
+				utils.AviLog.Warnf("key: %s, msg: Parent Reference %s of HTTPRoute object %s is not valid, err: %v", key, parentRefName, httpRoute.Name, err)
+			}
 		}
 	}
+
 	akogatewayapistatus.Record(key, httpRoute, &status.Status{HTTPRouteStatus: httpRouteStatus})
 
 	// No valid attachment, we can't proceed with this HTTPRoute object.
-	if invalidParentRefCount == len(httpRoute.Spec.ParentRefs) {
+	if invalidParentRefCount == len(httpRoute.Spec.ParentRefs) || !isValidHttprouteRules {
```

Comment:
GatewayAPI docs suggested instead of rejecting whole route , implementation should set route as partiallyInvalid.
https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPRouteRule 
```To reject an invalid combination or order of filters, implementations SHOULD consider the Route Rules with this configuration invalid. If all Route Rules in a Route are invalid, the entire Route would be considered invalid. If only a portion of Route Rules are invalid, implementations MUST set the “PartiallyInvalid” condition for the Route.```

Can we consider this?